otelconf/internal/provider/envprovider_test.go

```diff
@@ -0,0 +1,231 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package provider // import "go.opentelemetry.io/contrib/internal/provider"
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInvalidEnvVarName(t *testing.T) {
+	_, err := replaceEnvVar("$%&(*&)")
+	require.ErrorContains(t, err, errors.New("invalid environment variable name: $%&(*&)").Error())
+}
+
+func TestCheckRawConfTypeNil(t *testing.T) {
+	err := checkRawConfType([]byte{})
+	require.Error(t, err)
+	require.ErrorContains(t, err, "unsupported type=<nil> for retrieved config")
+}
+
+func TestReplaceEnvVar(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		setupEnv func(t *testing.T)
+		want     string
+		wantErr  bool
+	}{
+		{
+			name:  "no environment variables",
+			input: "key: value\nother: data",
+			want:  "key: value\nother: data",
+		},
+		{
+			name:     "simple environment variable substitution",
+			input:    "key: ${TEST_VAR}",
+			setupEnv: func(t *testing.T) { t.Setenv("TEST_VAR", "test_value") },
+			want:     "key: test_value",
+		},
+		{
+			name:  "undefined environment variable",
+			input: "key: ${UNDEFINED_VAR}",
+			want:  "key: ",
+		},
+		{
+			name:  "environment variable with default value",
+			input: "key: ${UNDEFINED_VAR:-default_value}",
+			want:  "key: default_value",
+		},
+		{
+			name:     "environment variable with default value when var is set",
+			input:    "key: ${DEFINED_VAR:-default_value}",
+			setupEnv: func(t *testing.T) { t.Setenv("DEFINED_VAR", "actual_value") },
+			want:     "key: actual_value",
+		},
+		{
+			name:  "escaped dollar sign - single escape",
+			input: "key: $${NOT_REPLACED}",
+			want:  "key: ${NOT_REPLACED}",
+		},
+		{
+			name:  "escaped dollar sign - double escape produces single dollar",
+			input: "key: $$${NOT_REPLACED}",
+			want:  "key: $",
```

Comment:
This value is replaced.

```suggestion
			input: "key: $$${TEST_VAR}",
			want:  "key: $test_value",
```